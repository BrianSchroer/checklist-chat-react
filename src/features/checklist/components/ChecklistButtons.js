import React, {PropTypes} from 'react';
import {connect} from 'react-redux';
import {bindActionCreators} from 'redux';
import {requestChecklistItemModalDialog} from '../../../app/modalDialogDucks';

class ChecklistButtons extends React.Component {
    constructor(props, context) {
        super(props, context);

        this.handleChecklistItemAddRequest  = this.handleChecklistItemAddRequest. bind(this);
    }

    handleChecklistItemAddRequest(event) {
        event.preventDefault();
        this.props.actions.requestChecklistItemModalDialog();
    }

    render() {
        return(
            <div>
                <button className="btn btn-primary checklist-item-button pull-right"
                    onClick={this.handleChecklistItemAddRequest}>Add Checklist Item...</button>
            </div>
        );
    }
}

ChecklistButtons.propTypes = {
    actions: PropTypes.object.isRequired
};

function mapStateToProps(state) {
    return state;
}

const mapDispatchToProps = (dispatch) => ({
    actions: bindActionCreators(
        {requestChecklistItemModalDialog},
        dispatch)
});

export default connect(mapStateToProps, mapDispatchToProps)(ChecklistButtons);
