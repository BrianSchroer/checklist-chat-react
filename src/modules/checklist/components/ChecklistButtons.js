import React from 'react';
import PropTypes from 'prop-types';

const ChecklistButtons = ({onChecklistItemAddRequest}) => {
    return(
        <div>
            <button className="btn btn-primary checklist-item-button pull-right"
                onClick={onChecklistItemAddRequest}>Add Checklist Item...</button>
        </div>
    );
};

ChecklistButtons.propTypes = {
    onChecklistItemAddRequest: PropTypes.func.isRequired
};

export default ChecklistButtons;
