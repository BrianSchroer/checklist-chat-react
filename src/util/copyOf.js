export default original => Object.assign({}, original);
