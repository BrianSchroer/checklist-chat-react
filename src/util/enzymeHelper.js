const enzymeHelper = class {

    static nestedFind(wrapper, selector) {
        const selectors = selector.split('>').map(node => node.trim());
        let parent = wrapper;
        let found = [];

        for (let i = 0; i < selectors.length; i++) {
            found = parent.find(selectors[i]);
            if (!found.length) {
                break;
            }
        }

        return found;
    }

    static findSingle(wrapper, selector) {
        const findArray = enzymeHelper.nestedFind(wrapper, selector);

        if (findArray.length != 1) {
            throw(`expected 1 find for selector "${selector}", but found ${findArray.length}.`);
        }

        return findArray.first();
    }
};

export default enzymeHelper;
