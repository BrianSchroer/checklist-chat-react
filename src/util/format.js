import moment from 'moment';

export default class {
    static time(input = '') {

        const isoDateFormatRegex = /\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}([+-]\d{2}:\d{2})?/g;
        let formatted = input;

        if (input && input.match(isoDateFormatRegex)) {
            formatted = this.dateTimeString(input, 'h:mm a');
        }

        return formatted;
    }

    static dateTimeString(input, formatString) {
        let formatted = input;

        if (input) {
            const testMoment = moment(input);

            if (testMoment.isValid()) {
                const testFormat = testMoment.format(formatString);
                if (testFormat != 'Invalid date') {
                    formatted = testFormat;
                }
            }
        }

        return formatted;
    }
}
