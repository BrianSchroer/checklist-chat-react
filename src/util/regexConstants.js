export const CSS_PATH = /(\.css)$/;
export const EOT_PATH = /\.eot(\?v=\d+\.\d+\.\d+)?$/;
export const JS_PATH = /\.js$/;
export const SVG_PATH = /\.svg(\?v=\d+\.\d+\.\d+)?$/;
export const TTF_PATH = /\.ttf(\?v=\d+\.\d+\.\d+)?$/;
export const WOFF_OR_WOFF2_PATH = /\.(woff|woff2)(\?v=\d+\.\d+\.\d+)?$/;
