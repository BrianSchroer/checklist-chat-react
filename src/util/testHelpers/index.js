import {shallow, mount} from 'enzyme';
import enzymeHelper from './enzymeHelper';

export {
    shallow as shallow,
    mount as mount,
    enzymeHelper as enzymeHelper
};
