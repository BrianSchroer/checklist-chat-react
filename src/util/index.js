import format from './format';
import uiHelper from './uiHelper';
import validationHelper from './validationHelper';

export {
    format as format,
    uiHelper as uiHelper,
    validationHelper as validationHelper
};
