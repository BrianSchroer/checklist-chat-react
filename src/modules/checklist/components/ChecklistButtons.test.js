import React from 'react';
import {snapshotHelper, shallow, enzymeHelper} from '../../../util/testHelpers';
import ChecklistButtons from './ChecklistButtons';

const defaultProps = {
    onChecklistItemAddRequest: () => {}
};

function overrideProps(propOverrides) {
    return Object.assign({}, defaultProps, propOverrides);
}

function shallowRender(propOverrides) {
    return shallow(<ChecklistButtons {...overrideProps(propOverrides)} />);
}

function assertSnapshotMatch(propOverrides = {}) {
    snapshotHelper.assertMatch(<ChecklistButtons {...overrideProps(propOverrides)} />);
}

describe('ChecklistButtons', () => {
    it('should render correctly', () => {
        assertSnapshotMatch();
    });

    it('should handle checklist item add request', () => {
        let wasCalled = false;

        const button = enzymeHelper.findSingle(
            shallowRender({ onChecklistItemAddRequest: () => {wasCalled = true;} }),
            'div > button.btn-primary');

        button.simulate('click');

        expect(wasCalled).toBe(true);
    });
});
