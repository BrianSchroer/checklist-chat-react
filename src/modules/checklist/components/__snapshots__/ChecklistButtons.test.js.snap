// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChecklistButtons should render correctly 1`] = `
<div>
  <button
    className="btn btn-primary checklist-item-button pull-right"
    onClick={[Function]}
  >
    Add Checklist Item...
  </button>
</div>
`;
